src/System.Management.Automation/engine/runtime/ScriptBlockToPowerShell.cs

```diff
@@ -382,16 +383,80 @@ private ScriptBlockToPowerShellConverter()
             return usingValueMap;
         }
 
+        // List of Foreach-Object command names and aliases.
+        // TODO: Look into using SessionState.Internal.GetAliasTable() to find all user created aliases.
+        //       But update Alias command logic to maintain reverse table that lists all aliases mapping
+        //       to a single command definition, for performance.
+        private static readonly string[] forEachNames = new string[]
+        {
+            "ForEach-Object",
+            "foreach",
+            "%"
+        };
+
+        private static bool FindForEachInCommand(CommandAst commandAst)
+        {
+            foreach (var commandElement in commandAst.CommandElements)
```

Comment:
I think you should only check on the first `CommandElement` since only that one is considered to be the command name. For example, I think a `CommandAst` representing `abc foreach -Parallel` will make this method return `true`.